### PR TITLE
added contexts to Celestia DataAvailabilityLayerClient

### LINKED ISSUE #354: Add context to DA interfaces
`context.Context` should be added as first argument to:
 * `SubmitBlock` and `CheckBlockAvailability` methods in [`DataAvailabilityLayerClient` go interface](https://github.com/celestiaorg/optimint/blob/7dae37342ffaffabc3d4ab23c027a2a0f0c99372/da/da.go#L57)
 * `RetrieveBlocks` method in [`BlockRetriever` go interface](https://github.com/celestiaorg/optimint/blob/7dae37342ffaffabc3d4ab23c027a2a0f0c99372/da/da.go#L75)

Changes also needs to be applied to `mock` and `grpc` implementations.

### PR DESCRIPTION
**Pull request by S1nus**
## Overview
Adding `Context` parameters to the `CheckBlockAvailability`, `SubmitBlock`, and `RetrieveBlocks` functions of the `DataAvailabilityLayerClient` object. Work-in-progress.

Goal is to enable requests to be cancelled easily, or allow them to have deadlines.

Resolves #354.

## Checklist
- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: da/celestia/celestia.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func (c *DataAvailabilityLayerClient) Stop() error {
 }
 
 // SubmitBlock submits a block to DA layer.
-func (c *DataAvailabilityLayerClient) SubmitBlock(block *types.Block) da.ResultSubmitBlock {
+func (c *DataAvailabilityLayerClient) SubmitBlock(context context.Context, block *types.Block) da.ResultSubmitBlock {
```
Comment (tzdybal) [inline da/celestia/celestia.go:65]:
```suggestion
func (c *DataAvailabilityLayerClient) SubmitBlock(ctx context.Context, block *types.Block) da.ResultSubmitBlock {
```

File: da/celestia/celestia.go [hunk 2]
```diff
@@ -103,7 +103,7 @@ func (c *DataAvailabilityLayerClient) SubmitBlock(block *types.Block) da.ResultS
 }
 
 // CheckBlockAvailability queries DA layer to check data availability of block at given height.
-func (c *DataAvailabilityLayerClient) CheckBlockAvailability(dataLayerHeight uint64) da.ResultCheckBlock {
+func (c *DataAvailabilityLayerClient) CheckBlockAvailability(context context.Context, dataLayerHeight uint64) da.ResultCheckBlock {
```
Comment (tzdybal) [inline da/celestia/celestia.go:106]:
```suggestion
func (c *DataAvailabilityLayerClient) CheckBlockAvailability(ctx context.Context, dataLayerHeight uint64) da.ResultCheckBlock {
```

**Review by tzdybal — CHANGES_REQUESTED** [review]
One comment, but applies in  multiple places - please rename variable from `context` to `ctx`.

File: da/mock/mock.go [hunk 3]
```diff
@@ -68,7 +69,7 @@ func (m *DataAvailabilityLayerClient) Stop() error {
 // SubmitBlock submits the passed in block to the DA layer.
 // This should create a transaction which (potentially)
 // triggers a state transition in the DA layer.
-func (m *DataAvailabilityLayerClient) SubmitBlock(block *types.Block) da.ResultSubmitBlock {
+func (m *DataAvailabilityLayerClient) SubmitBlock(context context.Context, block *types.Block) da.ResultSubmitBlock {
```
Comment (nashqueue) [inline da/mock/mock.go:72]: Same as above, quick change from `context` to `ctx`, as well for `CheckBlockAvailability` and `RetrieveBlocks`.

File: da/grpc/mockserv/cmd/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+cmd
+db
```
Comment (nashqueue) [inline da/grpc/mockserv/cmd/.gitignore:2]: [non-blocking] Why do we need cmd and db in gitignore ? How is that related to contexts.

**Review by nashqueue — CHANGES_REQUESTED** [review]

File: da/grpc/mockserv/cmd/.gitignore (hunk 4 above)
Comment (S1nus) [reply da/grpc/mockserv/cmd/.gitignore]: removed it

File: da/test/da_test.go [hunk 5]
```diff
@@ -172,7 +176,7 @@ func startMockCelestiaNodeServer(t *testing.T) *cmock.Server {
 	return httpSrv
 }
 
-func doTestRetrieve(t *testing.T, dalc da.DataAvailabilityLayerClient) {
+func doTestRetrieve(ctx context.Context, t *testing.T, dalc da.DataAvailabilityLayerClient) {
```
Comment (tzdybal) [inline da/test/da_test.go:179]: Actually, there is no point to pass context here - we can create `context.Background(()` inside this function (it makes more sense IMHO, as each test will have it's own context).

**Review by tzdybal — COMMENTED** [review]
Looks good. I left small comment regarding tests. Please rebase.

File: da/test/da_test.go [hunk 6]
```diff
@@ -62,14 +63,16 @@ func TestDALC(t *testing.T) {
...
 	for _, dalc := range registry.RegisteredClients() {
 		t.Run(dalc, func(t *testing.T) {
-			doTestDALC(t, registry.GetClient(dalc))
+			doTestDALC(ctx, t, registry.GetClient(dalc))
 		})
 	}
 }
 
-func doTestDALC(t *testing.T, dalc da.DataAvailabilityLayerClient) {
+func doTestDALC(ctx context.Context, t *testing.T, dalc da.DataAvailabilityLayerClient) {
```
Comment (tzdybal) [inline da/test/da_test.go:75]: Just like in `doTestRetrieve` - I think we can create context inside this method instead of passing it.

**Review by tzdybal — APPROVED** [review]
LGTM
